Add CI E2E testing for k8s-cluster-api-provider project (#534)
SovereignCloudStack/k8s-cluster-api-provider: doc/continuous-integration.md

  - It applies the PR label `cancelled-e2e-test` and leaves an informative PR comment when the `e2e-conformance` job is canceled

- `unlabel-on-update-e2e-test`
  - It is triggered by the PR update only when PR contains the `successful-e2e-test` label
  - It ensures that any PR update invalidates a previous successful e2e test
  - It removes `successful-e2e-test` label from the PR

- `e2e-quick-test`
  - It is triggered by the `e2e-quick-test` label in the opened PR
  - It executes `e2e-quick` job
  - It applies the PR label `successful-e2e-quick-test` and leaves an informative PR comment when the `e2e-quick` job succeeded
  - It applies the PR label `failed-e2e-quick-test` and leaves an informative PR comment when the `e2e-quick` job failed
  - It applies the PR label `cancelled-e2e-quick-test` and leaves an informative PR comment when the `e2e-quick` job is canceled

- `unlabel-on-update-e2e-quick-test`
  - It is triggered by the PR update only when PR contains the `successful-e2e-quick-test` label
  - It ensures that any PR update invalidates a previous successful e2e test
  - It removes `successful-e2e-quick-test` label from the PR

> 

Reviewer: I think it should be mentioned that the unlabel jobs are defined in the Zuul config repo?

Author: Find the sentence at the top of the `Pipelines` section:

`This section describes an [SCS Zuul pipelines](https://github.com/SovereignCloudStack/zuul-config/blob/main/zuul.d/gh_pipelines.yaml) that are used by the k8s-cluster-api-provider project.`

It already links the Zuul config repository as a source for all mentioned pipelines.

This secret is stored directly in the [SCS/zuul-config repository](https://github.com/SovereignCloudStack/zuul-config/blob/main/zuul.d/secrets.yaml) in an encrypted form.
It contains OpenStack application credentials to access the OpenStack project dedicated for CI testing.

This secret is encrypted by the SCS/zuul-config repository RSA key that has been generated by SCS Zuul instance.
So only SCS Zuul instance is able to decrypt it (read the [docs](https://zuul-ci.org/docs/zuul/latest/project-config.html#encryption)).

If you want to re-generate the mentioned secret or add another one using SCS/zuul-config repository RSA key, follow the below instructions:

- Install zuul-client
```bash
pip install zuul-client
```

- Encrypt "super-secret" string by the SCS/zuul-config repository public key from SCS Zuul
```bash
echo -n "super-secret" | \
  zuul-client --zuul-url https://zuul.scs.community encrypt \
  --tenant SCS \
  --project github.com/SovereignCloudStack/zuul-config
```

### FAQ

#### How do developers/reviewers should proceed if they want to CI test this project?

A developer initiates a PR as usual. If a reviewer deems that the PR requires e2e testing,

> he/she can apply a specific label to the PR.

Reviewer: ```suggestion
they can apply a specific label to the PR.
```
